logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -37,7 +37,9 @@ def config_name
     def multi_filter(events)
       @metric_events.increment(:in, events.size)
 
+      start = Time.now
       new_events = @filter.multi_filter(events)
+      @filter.metric.increment(:time_ms, ((Time.now - start)*1000).to_i)
```

Comment:
I didn't think about how the pipeline args could return accurate results with every snapshot, thanks for showing me the obvious here :)

I suggest a simple change, the metric lib has already a method called `#time` that create a `MeanType` instead of a counter. I suggest we change the `Metric#Time` to use a counter and remove the `MeanType`.

`#Time` accepts a block or will return a `TimedExecution` when no block is provided, we can call stop on it to actually record the time to the lib.

So this code could be rewritten like this.
```
new_events = @metric_events.time { @filter.multi_filter(events) }
```

We can use the `@metric_events` directly since it will be correctly scoped to the `plugins/events` node

Related code that need small changes.
https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/instrument/metric.rb#L35
https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/instrument/metric.rb#L84 We will need to change the value of this constant.